README.md

```diff
@@ -84,51 +58,45 @@ Run the recorder app in dev mode.
 npm run dev
 ```
 
-### Troubleshooting
-
-#### Page object is not defined
+#### Managing Playwright dependency
+When updating the version of [@elastic/synthetics](https://github.com/elastic/synthetics)(called _Synthetics agent_ hereafter), it is important to align the version of Playwright that the Synthetics agent uses and the forked Playwright that is installed by the recorder. Steps to update the Playwright is as followed:
 
-It is possible to define a series of steps that splits your recorded actions in a way that causes the required
-page object to be out of scope. You may be encountering this issue if, when testing or running your journey, you
-see an error message like:
+1. Go to [@elastic/playwright](https://github.com/elastic/playwright), fetch upstream microsoft:main into main if needed. We keep our modifications in `synthetics-recorder` branch. It is supposed to have only one extra commit[(84309bf)](https://github.com/elastic/playwright/commit/84309bf44d2a97889b178f2f2da2bc9f30e5aff8)) compared to the main branch. If main branch has new commits, fetch the changes into `synthetics-recorder` branch by pulling it with [rebase](https://git-scm.com/docs/git-pull#Documentation/git-pull.txt---rebasefalsetruemergesinteractive) option
 
-```javascript
-page1 is not defined
+1. Pull the remote changes to your machine. If necessary, set the remote as follows:
 ```
-
-You can verify the problem by clicking the `Export` button in the recorder and checking the code output. If your generated
-code is similar to the example below, you may need to rearrange your steps.
-
-```javascript
-step('Click text=Babel Minify', async () => {
-  // `page2` is defined here
-  const [page2] = await Promise.all([page.waitForEvent('popup'), page.click('text=Babel Minify')]);
-  await page2.click(':nth-match(a:has-text("babel-minify"), 3)');
-});
-step('Close page', async () => {
-  // referencing `page2`, which is now out of scope
-  await page2.close();
-});
+git remote add elastic https://github.com/elastic/playwright.git
+git remote -v
+// prints:
+// elastic  https://github.com/elastic/playwright.git (fetch)
+// elastic  https://github.com/elastic/playwright.git (push)
 ```
-
-By removing the second step divider in the Script Recorder UI, we will generate code that keeps the `page2` object
-in scope for all its references.
-
-```javascript
-step('Click text=Babel Minify', async () => {
-  // now all references to `page2` occur while it is still in scope
-  const [page2] = await Promise.all([page.waitForEvent('popup'), page.click('text=Babel Minify')]);
-  await page2.click(':nth-match(a:has-text("babel-minify"), 3)');
-  await page2.close();
-});
+2. Confirm the Playwright version from Synthetics agent's `package.json`, we will refer it as `v${NEW_VERSION}`. Checkout to `NEW_VERSION` tag, and create a new branch from there:
```

Comment:
This is confusing and assumes Synthetics agent PW version with the branch version which is not we are doing with the rebasing. Instead it would be the latest version of Microsoft playwright that was released. 

I do understand the intention here that we can keep a versioning scheme, But then we would have to assume the synthetics agent is always updated to the PW latest version.

Reply:
We might need to sync on this, but maybe we have a different approach on this.

Are you saying that the recorder should use the latest PW (let's say `1.24` or even higher that is not released) whereas the synthetics agent uses an older version(1.20.1 as of today)? I thought the intention was to use the same version of the playwright on both sides so we don't have any surprises. 

For this PR what I did was:
1. fetch `main` from upstream (PW latest version is 1.25.0-next)
2. create a dev branch that is sync up with the main and an extra commit with the modification for the recorder
3. Synthetics agent is using 1.20.1, so I checkout to `v1.20.1`, create a new branch named `1.20.1-recorder`, and then cherry-pick the commit from the dev branch
4. Use `1.20.1-recorder` branch from the recorder

But from my understanding, you are suggesting we do:
1. fetch `main` from upstream (PW latest version is 1.25.0-next)
2. create a dev branch that is sync up with the main and an extra commit with the modification for the recorder
3. Use dev branch from the recorder

Reply:
Myself and Kyungeun Synced on this separately, It was a misunderstanding in my part. The outcome is to update the docs with bit more detail so we can give more clarity around what versioning scheme we want to follow.

Reply:
I've updated the doc, let me know what you think! 🙏

Reply:
I've updated the doc, let me know what you think! 🙏